main.star

```diff
@@ -77,9 +77,17 @@ def run(args):
 			print("Waiting for the first finalized epoch")
 			first_cl_client = all_cl_client_contexts[0]
 			first_cl_client_id = first_cl_client.beacon_service_id
-			define_fact(service_id = first_cl_client_id, fact_name = FIRST_NODE_FINALIZATION_FACT, fact_recipe = struct(method= "GET", endpoint = "/eth/v1/beacon/states/head/finality_checkpoints", content_type = "application/json", port_id = HTTP_PORT_ID_FOR_FACT, field_extractor = ".data.finalized.epoch"))
-			finalized_epoch = wait(service_id = first_cl_client_id, fact_name = FIRST_NODE_FINALIZATION_FACT)
-			# TODO make an assertion on the finalized_epoch > 0
+			epoch_recipe = struct(
+				service_id = first_cl_client_id,
+				method= "GET",
+				endpoint = "/eth/v1/beacon/states/head/finality_checkpoints",
+				content_type = "application/json",
+				port_id = HTTP_PORT_ID_FOR_FACT,
+				extract = {
+					"finalized_epoch": ".data.finalized.epoch"
+				}
+			)
+			wait(epoch_recipe, "finalized_epoch", ">", 0)
```

Comment:
did you run this with waitForEpochFinalization turned to true?

Reply:
```
{
  "participants": [
    {
      "el_client_type": "geth",
      "el_client_image": "ethereum/client-go:v1.10.26",
      "cl_client_type": "lighthouse",
      "cl_client_image": "sigp/lighthouse:v3.3.0"
    },
    {
      "el_client_type": "geth",
      "el_client_image": "ethereum/client-go:v1.10.26",
      "cl_client_type": "prysm",
      "cl_client_image": "gcr.io/prysmaticlabs/prysm/beacon-chain:capella-post-merge,gcr.io/prysmaticlabs/prysm/validator:capella-post-merge"
    }
  ],
  "network_params": {
    "network_id": "3151908",
    "deposit_contract_address": "0x4242424242424242424242424242424242424242",
    "seconds_per_slot": 12,
    "slots_per_epoch": 32,
    "num_validator_keys_per_node": 64,
    "preregistered_validator_keys_mnemonic": "giant issue aisle success illegal bike spike question tent bar rely arctic volcano long crawl hungry vocal artwork sniff fantasy very lucky have athlete"
  },
  "wait_for_finalization": true,
  "wait_for_verifications": false,
  "global_client_log_level": "info"
}
```

Reply:
Doing it rn

Reply:
Worked!